Mathlib/SetTheory/Game/Birthday.lean

```diff
@@ -270,6 +270,52 @@ theorem birthday_sub_le (x y : Game) : (x - y).birthday ≤ x.birthday ♯ y.bir
 /- The bound `(x * y).birthday ≤ x.birthday ⨳ y.birthday` is currently an open problem. See
   https://mathoverflow.net/a/476829/147705. -/
 
+/-- Games with bounded birthday are a small set. -/
+theorem small_setOf_birthday_lt (o : Ordinal) : Small.{u} {x : Game.{u} // birthday x < o} := by
+  induction o using Ordinal.induction with | h o IH =>
+  let S := ⋃ a : Set.Iio o, {x : Game.{u} | birthday x < a}
+  have H : Small.{u} S := @small_iUnion _ _ _ _ (fun i => IH _ i.2)
+  obtain rfl | ⟨a, rfl⟩ | ho := zero_or_succ_or_limit o
+  · simp_rw [Ordinal.not_lt_zero]
+    exact small_empty
+  · simp_rw [Order.lt_succ_iff, le_iff_lt_or_eq]
+    convert @small_union.{u} _ {x | birthday x < a} {x | birthday x = a} _ _
+    · exact IH _ (Order.lt_succ a)
+    · let f : (S → Prop × Prop) → Game := fun g ↦ ⟦PGame.mk _ _
+        (fun x ↦ ((equivShrink {y // (g y).1}).symm x).1.1.out)
+        (fun x ↦ ((equivShrink {y // (g y).2}).symm x).1.1.out)⟧
+      suffices {x | x.birthday = a} ⊆ Set.range f from small_subset this
+      intro x hx
+      obtain ⟨y, hy, hy'⟩ := birthday_eq_pGameBirthday x
+      let g := fun z : S ↦ (∃ i, ⟦y.moveLeft i⟧ = z.1, ∃ j, ⟦y.moveRight j⟧ = z.1)
+      use g
+      rw [← hy]
+      apply PGame.game_eq <| PGame.Equiv.of_exists _ _ _ _ <;>
+      intro i
+      · obtain ⟨j, hj⟩ := ((equivShrink {y // (g y).1}).symm i).2
+        refine ⟨j, PGame.equiv_iff_game_eq.2 ?_⟩
+        rw [hj, PGame.moveLeft_mk, Quotient.out_eq]
+      · obtain ⟨j, hj⟩ := ((equivShrink {y // (g y).2}).symm i).2
+        refine ⟨j, PGame.equiv_iff_game_eq.2 ?_⟩
+        rw [hj, PGame.moveRight_mk, Quotient.out_eq]
+      · refine ⟨(equivShrink {y // (g y).1}) ⟨⟨⟦y.moveLeft i⟧, ?_⟩, i, rfl⟩, ?_⟩
+        · suffices ∃ b ≤ a, birthday ⟦y.moveLeft i⟧ < b from by simpa [S] using this
+          refine ⟨_, le_of_eq hx, ?_⟩
+          rw [← hy']
+          exact (birthday_quot_le_pGameBirthday _).trans_lt (PGame.birthday_moveLeft_lt i)
+        · rw [PGame.moveLeft_mk, Equiv.symm_apply_apply]
+          exact Quotient.mk_out _
+      · refine ⟨(equivShrink {y // (g y).2}) ⟨⟨⟦y.moveRight i⟧, ?_⟩, i, rfl⟩, ?_⟩
+        · suffices ∃ b ≤ a, birthday ⟦y.moveRight i⟧ < b from by simpa [S] using this
+          refine ⟨_, le_of_eq hx, ?_⟩
+          rw [← hy']
+          exact (birthday_quot_le_pGameBirthday _).trans_lt (PGame.birthday_moveRight_lt i)
+        · rw [PGame.moveRight_mk, Equiv.symm_apply_apply]
+          exact Quotient.mk_out _
+  · convert H
+    change birthday _ < o ↔ ∃ a, _
+    simpa using lt_limit ho
```

Comment:
```suggestion
  induction o using Ordinal.induction with | h o IH =>
  let S := ⋃ a ∈ Set.Iio o, {x : Game.{u} | birthday x < a}
  let H : Small.{u} S := @small_biUnion _ _ _ _ _ IH
  obtain rfl | ⟨a, rfl⟩ | ho := zero_or_succ_or_limit o
  · simp_rw [Ordinal.not_lt_zero]
    exact small_empty
  · simp_rw [Order.lt_succ_iff, le_iff_lt_or_eq]
    convert small_union.{u} {x | birthday x < a} {x | birthday x = a}
    · exact IH _ (Order.lt_succ a)
    · let f (g₁ g₂ : Set S) : Game := ⟦PGame.mk _ _
        (fun x ↦ ((equivShrink g₁).symm x).1.1.out) (fun x ↦ ((equivShrink g₂).symm x).1.1.out)⟧
      suffices {x | x.birthday = a} ⊆ Set.image2 f Set.univ Set.univ from small_subset this
      rintro x rfl
      obtain ⟨y, rfl, hy'⟩ := birthday_eq_pGameBirthday x
      let g₁ : Set S := {z | ∃ i, ⟦y.moveLeft i⟧ = z.1}
      let g₂ : Set S := {z | ∃ i, ⟦y.moveRight i⟧ = z.1}
      refine ⟨g₁, by simp, g₂, by simp, ?_⟩
      apply PGame.game_eq <| PGame.Equiv.of_exists _ _ _ _ <;>
      intro i
      · obtain ⟨j, hj⟩ := ((equivShrink g₁).symm i).2
        exact ⟨j, by simp [PGame.equiv_iff_game_eq, hj]⟩
      · obtain ⟨j, hj⟩ := ((equivShrink g₂).symm i).2
        exact ⟨j, by simp [PGame.equiv_iff_game_eq, hj]⟩
      · refine ⟨equivShrink g₁ ⟨⟨⟦y.moveLeft i⟧, ?_⟩, i, rfl⟩, by simp [Quotient.mk_out]⟩
        suffices ∃ b ≤ y.birthday, birthday ⟦y.moveLeft i⟧ < b by simpa [S, hy'] using this
        refine ⟨_, le_rfl, ?_⟩
        exact (birthday_quot_le_pGameBirthday _).trans_lt (PGame.birthday_moveLeft_lt i)
      · refine ⟨equivShrink g₂ ⟨⟨⟦y.moveRight i⟧, ?_⟩, i, rfl⟩, by simp [Quotient.mk_out]⟩
        suffices ∃ b ≤ y.birthday, birthday ⟦y.moveRight i⟧ < b by simpa [S, hy'] using this
        refine ⟨_, le_rfl, ?_⟩
        exact (birthday_quot_le_pGameBirthday _).trans_lt (PGame.birthday_moveRight_lt i)
  · convert H
    change birthday _ < o ↔ ∃ a, _
    simpa using lt_limit ho
```
Here's a golf, if you'd like

Reply:
Thanks! I took the opportunity to golf this even further ;)